app/src/main/java/org/schabi/newpipe/about/LicenseFragmentHelper.java

```diff
@@ -84,45 +80,31 @@ private static String getLicenseStylesheet(final Context context) {
      * @param color   the color number from R.color
      * @return a six characters long String with hexadecimal RGB values
      */
-    private static String getHexRGBColor(final Context context, final int color) {
+    private static String getHexRGBColor(@NonNull final Context context, final int color) {
         return context.getResources().getString(color).substring(3);
     }
 
-    @Nullable
-    private Activity getActivity() {
-        final Activity activity = weakReference.get();
-
-        if (activity != null && activity.isFinishing()) {
-            return null;
-        } else {
-            return activity;
-        }
-    }
-
-    @Override
-    protected Integer doInBackground(final Object... objects) {
-        license = (License) objects[0];
-        return 1;
-    }
-
-    @Override
-    protected void onPostExecute(final Integer result) {
-        final Activity activity = getActivity();
-        if (activity == null) {
+    static void showLicense(@Nullable final Context context, @NonNull final License license) {
+        if (context == null) {
             return;
         }
 
-        final String webViewData = Base64.encodeToString(getFormattedLicense(activity, license)
-                .getBytes(StandardCharsets.UTF_8), Base64.NO_PADDING);
-        final WebView webView = new WebView(activity);
-        webView.loadData(webViewData, "text/html; charset=UTF-8", "base64");
-
-        final AlertDialog.Builder alert = new AlertDialog.Builder(activity);
-        alert.setTitle(license.getName());
-        alert.setView(webView);
-        assureCorrectAppLanguage(activity);
-        alert.setNegativeButton(activity.getString(R.string.finish),
-                (dialog, which) -> dialog.dismiss());
-        alert.show();
+        Observable.fromCallable(() -> getFormattedLicense(context, license))
```

Comment:
Please also return this as a Disposable and add a CompositeDisposable in LicenseFragment